sdk/openai/azure-ai-openai-realtime/src/main/java/com/azure/ai/openai/realtime/RealtimeAsyncClient.java

```diff
@@ -0,0 +1,480 @@
+
+    // Server catch all:
+    private Sinks.Many<RealtimeServerEvent> serverEvents
+        = Sinks.many().multicast().onBackpressureBuffer(Queues.SMALL_BUFFER_SIZE, false);
+
+    /**
+     * Creates a RealtimeAsyncClient.
+     *
+     * @param webSocketClient The WebSocket client.
+     * @param cec The client endpoint configuration containing base URL, headers and other information.
+     * @param applicationId The application ID.
+     * @param retryStrategy The retry strategy. Unused for now
+     * @param authenticationProvider The authentication provider. Currently only supports KeyCredential and TokenCredential
+     */
+    RealtimeAsyncClient(WebSocketClient webSocketClient, ClientEndpointConfiguration cec, String applicationId,
+        RetryStrategy retryStrategy, AuthenticationProvider authenticationProvider) {
+        updateLogger(applicationId, null);
+
+        this.webSocketClient = webSocketClient == null ? new WebSocketClientNettyImpl() : webSocketClient;
+        this.clientEndpointConfiguration = cec;
+        this.applicationId = applicationId;
+        this.authenticationProvider = authenticationProvider;
+
+        this.sendMessageRetrySpec = Retry.from(signals -> {
+            AtomicInteger retryCount = new AtomicInteger(0);
+            return signals.concatMap(s -> {
+                Mono<Retry.RetrySignal> ret = Mono.error(s.failure());
+                if (s.failure() instanceof SendMessageFailedException) {
+                    if (((SendMessageFailedException) s.failure()).isTransient()) {
+                        int retryAttempt = retryCount.incrementAndGet();
+                        if (retryAttempt <= retryStrategy.getMaxRetries()) {
+                            ret = Mono.delay(retryStrategy.calculateRetryDelay(retryAttempt)).then(Mono.just(s));
+                        }
+                    }
+                }
+                return ret;
+            });
+        });
+    }
+
+    /**
+     * Closes the client.
+     */
+    @Override
+    public void close() {
+        if (this.isDisposed.getAndSet(true)) {
+            this.isClosedMono.asMono().block();
+        } else {
+            stop().then(Mono.fromRunnable(() -> {
+                this.clientState.changeState(RealtimeClientState.CLOSED);
+
+                isClosedMono.emitEmpty(emitFailureHandler("Unable to emit Close"));
+            })).block();
+        }
+    }
+
+    /**
+     * Starts the client for connecting to the server.
+     *
+     * @return the task.
+     */
+    public Mono<Void> start() {
+        return this.start(null);
+    }
+
+    /**
+     * Stops the client and disconnects from the server.
+     *
+     * @return the task.
+     */
+    public Mono<Void> stop() {
+        if (clientState.get() == RealtimeClientState.CLOSED) {
+            return Mono
+                .error(logger.logExceptionAsError(new IllegalStateException("Failed to stop. Client is CLOSED.")));
+        }
+
+        return Mono.defer(() -> {
```

Comment:
nit: declarative method calls should ideally not have large blocks of code. To improve readability, we should extract this into a method and here we should have `return Mono.defer(() -> doStop());`